mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,530 @@
+            // todo: verify if there is a possible deadlock here if i use states inside tear_down_bot
+            tear_down_bot(ctx).await;
+            break;
+        }
+        drop(states);
+        process_bot_logic(&ctx).await;
+        Timer::sleep(30.0).await;
+    }
+    info!("lp_bot_loop successfully stopped");
+}
+
+pub async fn process_price_request() -> Result<(StatusCode, String, HeaderMap), MmError<PriceServiceRequestError>> {
+    info!("Fetching price from: {}", KMD_PRICE_ENDPOINT);
+    let (status, headers, body) = slurp_url(KMD_PRICE_ENDPOINT).await?;
+    Ok((status, std::str::from_utf8(&body)?.trim().into(), headers))
+}
+
+async fn fetch_price_tickers(ctx: &MmArc) {
+    let (status_code, body, _) = match process_price_request().await {
+        Ok(x) => x,
+        Err(_) => return,
+    };
+    if status_code == StatusCode::OK {
+        let model: HashMap<String, TickerInfos> = match serde_json::from_str(&body) {
+            Ok(model) => model,
+            Err(_) => {
+                error!("error when unparsing the price fetching answer");
+                return;
+            },
+        };
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
+        let mut price_registry = simple_market_maker_bot_ctx.price_tickers_registry.lock().await;
+        price_registry.0 = model;
+        info!("registry size: {}", price_registry.0.len());
+    } else {
+        error!("error from price request: {} - {}", status_code, body);
+    }
+}
+
+pub async fn lp_price_service_loop(ctx: MmArc) {
+    info!("lp_price_service successfully started");
+    loop {
+        // todo: this log should probably in debug
+        info!("tick lp_price_service_loop");
+        if ctx.is_stopping() {
+            break;
+        }
+
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+        let states = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        if *states == TradingBotState::Stopping {
+            info!("stop price service loop");
+            break;
+        }
+        drop(states);
+        fetch_price_tickers(&ctx).await;
+        Timer::sleep(20.0).await;
+    }
+    info!("lp_price_service successfully stopped");
+}
+
+pub async fn start_simple_market_maker_bot(ctx: MmArc, req: StartSimpleMakerBotRequest) -> StartSimpleMakerBotResult {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+    {
+        let mut states = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        if *states == TradingBotState::Running {
+            return MmError::err(StartSimpleMakerBotError::AlreadyStarted);
+        }
+        let mut trading_bot_cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
+        *trading_bot_cfg = req.cfg;
+        *states = TradingBotState::Running;
+    }
+
+    info!("simple_market_maker_bot successfully started");
+    spawn(lp_price_service_loop(ctx.clone()));
```

Comment:
It seems that `lp_price_service_loop` and `lp_bot_loop` should be in the same loop. It will help to avoid sharing additional state (TickerInfosRegistry), that is kept the same in case of error `fetch_price_tickers` - it can potentially lead to more errors in other places.